language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureInfoModelBuilder.java

```diff
@@ -0,0 +1,342 @@
+            if (firstIncludedRecordParam().isPresent() && !paramModel.isIncludedRecordParam()) {
+                expandedSignatureParamInfo.add(paramInfo);
+            }
+        }
+
+        if (firstIncludedRecordParam().isPresent()) {
+            int firstIncludedRecordParamIndex = this.parameterModels.indexOf(firstIncludedRecordParam().get());
+            List<ParameterInfoModel> includedRecordParams = this.parameterModels.stream()
+                    .filter(ParameterInfoModel::isIncludedRecordParam)
+                    .collect(Collectors.toList());
+
+            StringBuilder expandedLabelBuilder = new StringBuilder(expandedModelPrefix.toString());
+            for (int i = 0; i < includedRecordParams.size(); i++) {
+                int labelOffset = expandedLabelBuilder.toString().length();
+                ParameterInfoModel includedParamInfo = includedRecordParams.get(i);
+                TypeSymbol paramTypeSymbol = includedParamInfo.getParameter().getParameterSymbol().typeDescriptor();
+                Pair<String, List<ParameterInformation>> includedRecordParamInfo =
+                        getIncludedRecordParamInfo(paramTypeSymbol, labelOffset);
+                expandedLabelBuilder.append(includedRecordParamInfo.getKey());
+
+                if (firstIncludedRecordParamIndex + i < parameterModels.size() - 1) {
+                    expandedLabelBuilder.append(", ");
+                }
+
+                expandedSignatureParamInfo.addAll(includedRecordParamInfo.getValue());
+            }
+            expandedLabelBuilder.append(")");
+            result.add(getSignatureInfo(expandedLabelBuilder.toString(), expandedSignatureParamInfo));
+        }
+
+        defaultLabelBuilder.append(")");
+        result.add(getSignatureInfo(defaultLabelBuilder.toString(), defaultSignatureParamInfo));
+
+        return result;
+    }
+
+    private SignatureInformation getSignatureInfo(String label, List<ParameterInformation> parameters) {
+        SignatureInformation expandedSignatureInfo = new SignatureInformation();
+        expandedSignatureInfo.setDocumentation(this.signatureDescription);
+        expandedSignatureInfo.setLabel(label);
+        expandedSignatureInfo.setParameters(parameters);
+        return expandedSignatureInfo;
+    }
+
+    private void fillParamInfoModels() {
+        fillDescriptionModels();
+        Optional<FunctionTypeSymbol> functionTypeSymbol = getFunctionType(symbol);
+        List<ParameterSymbol> parameterSymbols = functionTypeSymbol
+                .map(typeSymbol -> typeSymbol.params().orElse(new ArrayList<>()))
+                .orElse(Collections.emptyList());
+        if (parameterSymbols.isEmpty()) {
+            return;
+        }
+
+        List<Parameter> parameters = parameterSymbols
+                .subList(skipFirstParam() ? 1 : 0, parameterSymbols.size())
+                .stream()
+                .map(param -> new Parameter(param, false, false, context))
```

Comment:
Shall we introduce a new constructor in `Parameter` class which sets the `isOptional` and `isRest `flags to false.

Reply:
Since this is a constructor usage, having two default fields might not be descriptive since the default values of the two flags can be opinionated. WDYT?